File: Typescript/src/minCosts.ts
export function minCosts(cost: number[]): number[] {
	let min = cost[0];
	let minCosts = Array(cost.length);
	minCosts[0] = min;
	for (let i = 1; i < cost.length; i++) {
		if (cost[i] < min) {
			min = cost[i];
		}
		minCosts[i] = min;
	}
	return minCosts;
};

File: Dart/lib/minCosts.dart
import 'dart:math';

class Solution {
  List<int> minCosts(List<int> cost) {
    if (cost.isEmpty) return [];
    List<int> result = [cost[0]];
    var costMin = cost[0];
    for (int i = 1; i < cost.length; i++) {
      costMin = min(cost[i], costMin);
      result.add(costMin);
    }
    return result;
  }
}
